crates/storage/nippy-jar/src/compression/mod.rs

```diff
@@ -37,35 +43,60 @@ pub trait Compression: Serialize + for<'a> Deserialize<'a> {
 #[cfg_attr(test, derive(PartialEq))]
 pub enum Compressors {
     Zstd(Zstd),
+    Lz4(Lz4),
     // Avoids irrefutable let errors. Remove this after adding another one.
     Unused,
 }
 
 impl Compression for Compressors {
+    fn decompress_to(&self, value: &[u8], dest: &mut Vec<u8>) -> Result<(), NippyJarError> {
+        match self {
+            Compressors::Zstd(zstd) => zstd.decompress_to(value, dest),
+            Compressors::Lz4(lz4) => lz4.decompress_to(value, dest),
+            Compressors::Unused => unimplemented!(),
+        }
+    }
     fn decompress(&self, value: &[u8]) -> Result<Vec<u8>, NippyJarError> {
         match self {
             Compressors::Zstd(zstd) => zstd.decompress(value),
+            Compressors::Lz4(lz4) => lz4.decompress(value),
             Compressors::Unused => unimplemented!(),
         }
     }
 
-    fn compress_to<W: Write>(&self, src: &[u8], dest: &mut W) -> Result<(), NippyJarError> {
-        match self {
-            Compressors::Zstd(zstd) => zstd.compress_to(src, dest),
-            Compressors::Unused => unimplemented!(),
+    fn compress_to(&self, src: &[u8], dest: &mut Vec<u8>) -> Result<usize, NippyJarError> {
+        let initial_capacity = dest.capacity();
+        loop {
+            let result = match self {
+                Compressors::Zstd(zstd) => zstd.compress_to(src, dest),
+                Compressors::Lz4(lz4) => lz4.compress_to(src, dest),
+                Compressors::Unused => unimplemented!(),
+            };
+
+            match result {
+                Ok(v) => return Ok(v),
+                Err(err) => match err {
+                    NippyJarError::OutputTooSmall => {
+                        dest.reserve(initial_capacity);
```

Comment:
I see, so we extend if we the capacity is too small

makes sense

Reply:
it's there in a `just in case`. it shouldn't happen though